nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/header/components/nf-ng-group-component.js

```diff
@@ -129,17 +133,78 @@
                  * Initialize the modal.
                  */
                 init: function () {
+                    var self = this;
+
+                    self.fileForm = $('#file-upload-form').ajaxForm({
+                        url: '../nifi-api/process-groups/',
+                        dataType: 'json',
+                        beforeSubmit: function ($form, options) {
+                            // ensure uploading to the current process group
+                            options.url += (encodeURIComponent(nfCanvasUtils.getGroupId()) + '/process-groups/upload');
+                        }
+                    });
+
                     // configure the new process group dialog
                     this.getElement().modal({
                         scrollableContentStyle: 'scrollable',
                         headerText: 'Add Process Group',
                         handler: {
                             close: function () {
+                                self.fileToBeUploaded = null;
+                                $('#selected-file-name').text('');
+                                $('#upload-file-field').val('');
                                 $('#new-process-group-name').val('');
                                 $('#new-process-group-dialog').removeData('pt');
+
+                                // reset the form to ensure that the change fire will fire
+                                self.fileForm.resetForm();
                             }
                         }
                     });
+
+                    $('#upload-file-field-button').on('click', function (e) {
+                        $('#upload-file-field').click();
+                    });
+
+                    $('#upload-file-field').on('change', function (e) {
+                        $('#upload-file-field-button').hide();
+
+                        self.fileToBeUploaded = e.target;
+                        var filename = $(this).val();
+                        var filenameExtension;
+                        if (!nfCommon.isBlank(filename)) {
+                            filenameExtension = filename.replace(/^.*[\\\/]/, '');
+                            filename = filename.replace(/^.*[\\\/]/, '').replace(/\..*/, '');
```

Comment:
Should these regular expression replacements be moved into a separate function that describes what is happening?  Otherwise the purpose is not quite clear.

Reply:
Revised - I moved to two separate methods describing its functions.